scopes/component/ui/component-meta/component-overview.tsx

```diff
@@ -44,30 +53,59 @@ export function ComponentOverview({
     const origin = isBrowser ? window.location.origin : undefined;
     finalElementsUrl = origin && elementsUrl ? `${origin}${elementsUrl}` : undefined;
   }
+
+  type CompWithTitleBadgesProps = { position: TitleBadgePosition; children: ReactNode };
+  function ComponentWithTitleBadges({ position, children }: CompWithTitleBadgesProps) {
+    return (
+      <div className={styles.componentWithBadgeContainer}>
+        {children}
+        <div className={styles.badgeContainer}>
+          {titleBadges
+            // eslint-disable-next-line react/prop-types
+            ?.filter((badge) => {
+              return (
+                (position === TitleBadgePosition.Title && !badge.position) || // default position is title
+                badge.position === position
+              );
+            })
+            // @ts-ignore
+            ?.sort((a, b) => a?.weight - b?.weight)
+            ?.map((titleBadge, index) => {
+              return (
+                <titleBadge.component
+                  key={index}
+                  componentDescriptor={componentDescriptor}
+                  legacyComponentModel={component}
+                />
+              );
+            })}
```

Comment:
memo this (I know it already exists in the original code, but it needs memo-ing)